### PR TITLE
Make all Virtual console games 4:3 (and cleanup GameInis)

### PR DESCRIPTION
**Pull request by phire**
### Part 1

Since all Virtual console releases use the same emulator, all the releases for a given console generally need the same setting, this PR allows the creation of INI files which cover all Virtual Console games for each system.

For example, F.ini can provide settings that all NES games have in common and C.ini will provide settings for all Commodore 64 games.

### Part 2

Might as well de-duplicate all the files while we are here, this gets rid of over 300 un-needed ini files.

### Part 3

And then the actual reason why I did this: By simply changing 9 files, all virtual console game will now automatically select the correct (non-widescreen) aspect ratio. This fixes [Issue 9243](https://bugs.dolphin-emu.org/issues/9243) and stops people from incorrectly complaining that my double strike aspect ratio fixes actually don't fix things.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3591)

### CODE REVIEW

**Comment by phire** [comment]
Could @JMC47 and @Linktothepast look over this, it touches a lot of ini files?

Also, Can we get this merged into 5.0 @delroth? Most virtual console games now almost perfectly and it's would be a shame to have them all at the wrong aspect ratio.

**Comment by Linktothepast** [comment]
I have some objections about this. If the general setting for wii is set to 16:9 why should we enforce a general setting for virtual console ourselves? I don't mind if you are given the option to change it easily for all the virtual console games of a console by changing a single file but i don't like the dictation of a setting which is a pure matter of choice. Sure the original games were 4:3 but when the console itself was giving you the choice between 16:9 and 4:3 why should we pick for the user?

Also some games needed different settings (like efb to ram or xfb in certain genesis and nes games for example), and seeing certain mistakes in the inis by a quick look i am not so confident about the conversion. I will comment on the mistakes afterwards.

File: Data/Sys/GameSettings/FAA.ini [hunk 1]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAA.ini]: These lines are unnecessary and do nothing.

File: Data/Sys/GameSettings/FAB.ini [hunk 2]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAB.ini]: Same here, unnecessary lines that do nothing.

File: Data/Sys/GameSettings/FAG.ini [hunk 3]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAG.ini]: Same here.

File: Data/Sys/GameSettings/FAH.ini [hunk 4]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAH.ini]: Same here.

File: Data/Sys/GameSettings/FAL.ini [hunk 5]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAL.ini]: Same here.

File: Data/Sys/GameSettings/FAN.ini [hunk 6]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAN.ini]: Same here.

File: Data/Sys/GameSettings/FAO.ini [hunk 7]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 512
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAO.ini]: Same here.

File: Data/Sys/GameSettings/FAS.ini [hunk 8]
```diff
@@ -1,26 +1,4 @@
...
-# Add action replay cheats here.
-
-[Video_Settings]
-SafeTextureCacheColorSamples = 0
-EFBScale = 2
-
-[Video_Enhancements]
-MaxAnisotropy = 0
-ForceFiltering = False
+[Video_Hacks]
```
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAS.ini]: Same here.

File: Data/Sys/GameSettings/FAA.ini (hunk 1 above)
Comment (phire) [inline Data/Sys/GameSettings/FAA.ini]: Not true.

They override the `EFBToTextureEnable = False` line in `F.ini` so dolphin will use either Efb2ram or Efb2tex depending on what the user has set in their settings.

**Comment by fallaha56** [comment]
@Linktothepast re: enforcing: I can see why you might want a single setting either in the GUI or .ini for VC games but surely it's obvious that folk will almost always want Wii games in 16:9 but VC games in corrected 4:3?

File: Data/Sys/GameSettings/FAA.ini (hunk 1 above)
Comment (Linktothepast) [inline Data/Sys/GameSettings/FAA.ini]: Ah ok, sorry my bad, the line was hidden at the online viewer and didn't notice it. You can disregard all these comments.

**Comment by phire** [comment]
> If the general setting for wii is set to 16:9 why should we enforce a general setting for virtual console ourselves?

This PR was actually inspired by the behavior of vWii mode on the Wii-U, which (after an update) would automatically force any title ID with the first letter matching one of [C, E, F, J, L, M, N, P, Q] into 4:3 pillerboxed mode.

The reason why the original Wii doesn't support this feature is it has no way of signaling the aspect ratio to the TV, relying on the user to manually switch the aspect ratio setting on their remote. I assume if Nintendo had the ablity to signal aspects on the Wii, they would have added this functionality in an update.

> Sure the original games were 4:3 but when the console itself was giving you the choice between 16:9 and 4:3 why should we pick for the user?

Technically it's not the console which gives you a choice, it's the user's TV. The Wii really has no idea what 16:9 even is.

This PR still allows the user to override the aspect ratio and get 16:9 stretched Virtual Console titles, if they really want it. There are actually two methods:

 * Right click the game, properties, check "enable widescreen"
 * Graphics options, Set aspect ratio to "Force 16:9"

This is the reason why I used `Widescreen = False` instead of `AspectRatio = 2` because it allows overriding (and doesn't switch aspect ratios when you open the graphics tab)

**Comment by Linktothepast** [comment]
So you are saying that is more of a console limitation rather than a choice made by nintendo? And ok i agree you can easily override it if you wanted. If most people and devs like this change i won't raise any further objections.

**Comment by JosJuice** [comment]
I like this PR. It's pretty much exactly how I would've wanted to do this.

---

Reviewed 374 of 374 files at r1.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gnwRFI2bgOG0cPjUK:-K9gnwRFI2bgOG0cPjUL:1558041232) (no related file):*
Should we also set `Widescreen = False` for Wii games that only can run in 4:3, like WarioWare: Smooth Moves?

---

*[Data/Sys/GameSettings/C.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9goJOuSime8DA2V-6t:-K9goJOuSime8DA2V-6u:-243213492) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/C.ini#L1)):*
Virtual Console (capitalization), Commodore (spelling).

---

*[Data/Sys/GameSettings/E.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gopBxbMed658iWkj1:-K9gopBysKr5dT7mFePl:-826535508) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/E.ini#L1)):*
Virtual Console should come after Arcade so that it's consistent with the other systems. NeoGeo Virtual Console needs to have a capital C.

---

*[Data/Sys/GameSettings/FAA.ini, line 3 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9fj4SjQQ4h0XpU_rpH-r1-3:-K9gpixmiH5YyDGcmga6:-1326105924) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/FAA.ini#L3)):*
Is it really true that this game doesn't need EFB to RAM? (And the other ones below)

---

*[Data/Sys/GameSettings/J.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gqYO59SKmHFn2mHWr:-K9gqYO6fRrkAMQ59Nyt:-1021639662) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/J.ini#L1)):*
Needs to have "games" in lowercase to match the other INIs.

---

*[Data/Sys/GameSettings/L.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9grgJ3iScdVKDqKHhk:-K9grgJ4tbtJsHFU0aBV:-631915738) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/L.ini#L1)):*
Remove the period for consistency.

---

*[Data/Sys/GameSettings/M.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gsI9pc19hJJHb52D4:-K9gsI9q5yyw8JUMYqBa:-1276596029) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/M.ini#L1)):*
Needs to be Sega Genesis, both here and in the commit title. (Or Sega Mega Drive if you want to use the proper name ;))

---

*[Data/Sys/GameSettings/N.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9guAdMc98YuI3nAUEn:-K9guAdNioZQ-D0MMfTw:-11157739) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/N.ini#L1)):*
Typo of Virtual.

---

*[Data/Sys/GameSettings/P.ini, line 2 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9guXLPA5rvuXVjO3hj:-K9guXLPA5rvuXVjO3hk:-1814970828) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/P.ini#L2)):*
Shouldn't this be capitalized TurboGrafx (both here and in the commit title)? Also, GameCube needs to have a capital C here, and the word Virtual is misspelled in the two last commit titles.

---

*[Data/Sys/GameSettings/PZLE01.ini, line 647 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gvfeqAAkm_8vB306R:-K9gvfeqAAkm_8vB306S:42998521) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/PZLE01.ini#L647)):*
Six-character INIs should only contain patches and cheat codes. Put this in PZL.ini instead.

---

*[Data/Sys/GameSettings/Q.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gwscZbPMPDTuCBXvQ:-K9gwsc_rgSG7xW_yEvb:24241546) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/Q.ini#L1)):*
Again, should this be capitalized TurboGrafx here and in the commit title?

---

*[Source/Core/Core/ConfigManager.cpp, line 976 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gxg6lNibFn2-QkT0_:-K9gxg6lNibFn2-QkT0a:757643669) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Source/Core/Core/ConfigManager.cpp#L976)):*
Space after the `if`

---

*[Source/Core/Core/ConfigManager.cpp, line 979 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gySAnZ8VTnriqIJ5Y:-K9gySAoffpodPWn108O:115705237) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Source/Core/Core/ConfigManager.cpp#L979)):*
This would look nicer if "system" and "unique" are lowercase, but it doesn't matter much.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-:-K9gnoaODzLhqqnVwtmT:472126986)*

**Comment by JMC47** [comment]
I'm worried about multiple revisions of the emulators, but based on the settings I'm seeing all the games should work.

**Comment by mbc07** [comment]
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gnwRFI2bgOG0cPjUK:-K9hDa28MtmKXkXhjUHs:947888393) (no related file):*
+1 on this. I didn't know about `Widescreen = False` and it seems very promising for a few early Wii titles that works only at 4:3 mode but display stretched by using Dolphin's Auto AR with Wii settings set to 16:9...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591)*

**Comment by phire** [comment]
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gnwRFI2bgOG0cPjUK:-K9h5KvPQTSusITcP2hf:650281586) (no related file):*
That would be a good idea. 

Do you want to submit a PR?

---

*[Data/Sys/GameSettings/E.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gopBxbMed658iWkj1:-K9h8DPBu9nCSr5alWOB:1488021112) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/E.ini#L1)):*
They are actually listed as the "Virtual Console Arcade" platform in the wii shop. To follow the pattern would result in "All Virtual Console Arcade Virtual Console games...."

---

*[Data/Sys/GameSettings/FAA.ini, line 3 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9fj4SjQQ4h0XpU_rpH-r1-3:-K9h8XRWc_O8reajKFy_:166187963) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/FAA.ini#L3)):*
There are different versions of the NES emulator and game might be using different features. I notice that all the games without EFB to RAM are simpler games, most of them don't have mappers.

---

*[Data/Sys/GameSettings/P.ini, line 2 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9guXLPA5rvuXVjO3hj:-K9h6bo_9ziabvxGAHIb:1856034118) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/P.ini#L2)):*
The list I was following had it capitalized as TurboGraFX, but I can't find any other examples of it being capitalized in that way.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591)*

**Comment by mbc07** [comment]
Review status: 364 of 374 files reviewed at latest revision, 20 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gnwRFI2bgOG0cPjUK:-K9jjZP9AVrWNxUvTZSO:2013284757) (no related file):*
Sounds good. I'll try to gather other Wii titles that runs at 4:3 (I know there are more) before submitting a new PR. If I can't find a reliable source I'll end updating only WarioWare Smooth Moves though...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591)*

**Comment by JMC47** [comment]
Harvest Moon: Animal Parade is a 4:3 only Wii game just to jump start you.

As for this PR, Master System games need RealXFB to render correctly.

**Comment by phire** [comment]
@Linktothepast after some testing, I've set all Sega Master System games to RealXFB. 

They all do the same strided XFB copies as Sonic and Knuckles, which is currently only implemented correctly in RealXFB. Virtual XFB and XFB disabled both end up with the wrong aspect ratio and the right side of the screen missing.

It should be possible in the future to get these strided XFB copies working with Virtual XFB too.

**Comment by JosJuice** [comment]
Reviewed 9 of 10 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[Data/Sys/GameSettings/E.ini, line 1 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9gopBxbMed658iWkj1:-K9kdTO_IdhA9PQVMcPs:-1024557721) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/E.ini#L1)):*
Fair enough.

---

*[Data/Sys/GameSettings/P.ini, line 2 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9guXLPA5rvuXVjO3hj:-K9kezyDNJUPMJ4WpZOp:1450911931) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/P.ini#L2)):*
GameCube still needs a capital C here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591)*

**Comment by phire** [comment]
Review status: 363 of 384 files reviewed at latest revision, 1 unresolved discussion.

---

*[Data/Sys/GameSettings/FAAE01.ini, line 3 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591#-K9fj4SjQQ4h0XpU_rpH-r1-3:-K9l-rxOk_fipbcbai8Q:47456159) ([raw file](https://github.com/dolphin-emu/dolphin/blob/0f7506b6cbfded624067eaf0764670f9abdfbfa4/Data/Sys/GameSettings/FAAE01.ini#L3)):*
I spent some time going over which NES games didn't need EFB2Ram with JMC47. Turns out the old INIs weren't quite correct because they assumed every region used the same version of emulator in it's release, which turns out not to be true. Both Donkey Kong and Super Mario Bros need EFB2Ram in the PAL release and not the US release.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3591)*

**Comment by Linktothepast** [comment]
Interesting has anyone tested what happens with japanese nes games?

**Comment by Linktothepast** [comment]
I would just like to add that the builder bot needs a clean up of the gameini folder again due to the removal of 300 ini files that still show up in dolphin builds.

**Pull request merged**